File: prob19.js
// https://projecteuler.net/problem=19
// Completed 3/20/2022

let numOfSundays = 0;

let daysOfWeek = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
// First day of 1901 was a Tuesday
let daysOfWeekIndex = 1;


for(let year = 1901; year <= 2000; year++){
    for(let month = 1; month <= 12; month++){

        if(month === 2){
            if(daysOfWeek[daysOfWeekIndex] === 'Sunday'){
                numOfSundays++;
            }
            // Checks if it is a leap year
            if(year % 4 === 0){
                // Adds rest of days of the month since only first is looked at (applied to rest of months)
                daysOfWeekIndex += 29
                daysOfWeekIndex %= 7;
                }else{
                daysOfWeekIndex += 28
                daysOfWeekIndex %= 7;
                }
        } else if(month === 1 || month === 3 || month === 5 || month === 7 || month === 8 || month === 10 || month === 12){
            if(daysOfWeek[daysOfWeekIndex] === 'Sunday'){
                numOfSundays++;
            }
            daysOfWeekIndex += 31
            daysOfWeekIndex %= 7;
        } else {
            if(daysOfWeek[daysOfWeekIndex] === 'Sunday'){
                numOfSundays++;
            }
            daysOfWeekIndex += 30
            daysOfWeekIndex %= 7;
        }
    }
}

console.log(numOfSundays)

File: prob19.py
# https://projecteuler.net/problem=19
# Completed 3/20/2022

def daysInMonth():
    if month in [1,3,5,7,8,10,12]:
        return 31
    elif month in [4,6,9,11]:
        return 30
    else:
        if year % 4 == 0 and (year % 100 != 0 or year % 400 == 0):
            return 29
        return 28

count = 0;
weekday = 2

for year in range(1900, 2001):
    for month in range(1,13):
        for day in range(1, daysInMonth()+1):
            if day == 1 and weekday == 1 and year > 1900:
                count += 1
            if weekday < 7:
                weekday +=1
            else:
                weekday = 1
                
print(count)